### DISCUSSION TITLE
[Proposal] Monitoring mixins

### DISCUSSION THREAD

**Discussion by P6rguVyrst** [gravitational/teleport#16150] [Ideas]
I like when software comes with batteries included. 

Whilst it's amazing that Teleport comes with Prometheus style  instrumentation, it would be even greater if it came along with alerts, recording rules and dashboards out of the box. This would help concentrate developer intent with operational knowledge of people running the software, and develop alerting and dashboards that crowdsource all that experience. 

I'd be happy to volunteer for this, if this kind of contribution would be welcome.

**Examples:**
https://github.com/thanos-io/thanos/tree/master/mixin
https://github.com/kubernetes-monitoring/kubernetes-mixin
https://github.com/prometheus/node_exporter/tree/master/docs/node-mixin

I believe the monitoring mixin should live alongside code instrumentation, which would be alongside teleport codebase.

**Comment by thameezb** [comment]
This would be a great initiative
